include/boost/test/impl/execution_monitor.ipp

```diff
@@ -242,6 +242,9 @@ extract( boost::exception const* ex )
 //____________________________________________________________________________//
 
 static void
+#ifdef __GNUC__
+__attribute__((__format__ (__printf__, 3, 0)))
```

Comment:
How can we ensure that this attribute is available for the target compiler?

Reply:
I'll change the check to `__GNUC__ >= 3`. Or would you prefer to set this only for specific compilers (I've tested with gcc 6-8 and clang 5-9)?